runtime/lua/vim/lsp.lua

```diff
@@ -1030,7 +1043,10 @@ function lsp.stop_client(client_id, force)
     if resolved_client_id then
       local client_buffers = lsp.get_buffers_by_client_id(resolved_client_id)
       for idx = 1, #client_buffers do
-        lsp.diagnostic.clear(client_buffers[idx], resolved_client_id)
+        lsp.diagnostic.clear_diagnostic_cache(client_buffers[idx], resolved_client_id)
```

Comment:
seems like we could call `client.stop()` here at a first glance. Why can't we do that?

It looks pretty copy & pasted to me.

Reply:
You mean replace stop_client with a wrapper around client.stop()? I don't see why not.

Reply:
Oh, I'm dumb. The issue is my fault since I should have just implemented this behavior in client.stop in my earlier PR. fixed